libafl/src/stats/mod.rs

```diff
@@ -204,3 +236,440 @@ where
         }
     }
 }
+
+#[macro_export]
+macro_rules! start_timer {
+    ($state:expr) => {{
+        // Start the timer
+        #[cfg(feature = "perf_stats")]
+        $state.perf_stats_mut().start_timer();
+    }};
+}
+
+#[macro_export]
+macro_rules! mark_feature_time {
+    ($state:expr, $feature:expr) => {{
+        // Mark the elapsed time for the given feature
+        #[cfg(feature = "perf_stats")]
+        $state.perf_stats_mut().mark_feature_time($feature);
+    }};
+}
+
+#[macro_export]
+macro_rules! mark_feedback_time {
+    ($state:expr) => {{
+        // Mark the elapsed time for the given feature
+        #[cfg(feature = "perf_stats")]
+        $state.perf_stats_mut().mark_feedback_time();
+    }};
+}
+
+/// Number of stages in the fuzzer
+const NUM_STAGES: usize = 4;
+
+/// Number of feedback mechanisms to measure for performance
+pub const NUM_FEEDBACKS: usize = 4;
```

Comment:
FWIW Just tried it, but `ST::LEN` doesn't work yet everywhere. It's not available at some places and we would need nightly features at others. We could use `Vec`s instead of arrays, though. Not sure if/how much this would impact perf.
Or just leave it as-is...

Reply:
The arrays are there to ensure we don't do any allocations. I don't mind either way really. We could always just do a vec::with_capacity I guess